net_ipc/src/socket.rs

```diff
@@ -50,6 +58,7 @@ impl IpcSocket for ZmqIpcSocket {
 
     fn connect(&mut self, endpoint: &str) -> Result<()> {
         self.socket.connect(endpoint)?;
+        self.maybe_endpoint = Some(endpoint.to_string());
```

Comment:
rather than regurgitating what was passed in, it may make sense to return the endpoint that was actually connected: https://docs.rs/zmq/0.9.0/zmq/struct.Socket.html#method.get_last_endpoint

Reply:
done